packages/integration-sdk-runtime/src/execution/__tests__/jobState.test.ts

```diff
@@ -105,8 +136,133 @@ describe('#findEntity', () => {
+    await jobState.addEntities([e2]);
+
+    const expectedUploaded: FlushedGraphObjectData[] = [
+      {
+        entities: [e1, e2],
+        relationships: [],
+      },
+    ];
+
+    expect(uploadCollector.graphObjectDataCollection).toEqual(expectedUploaded);
+  });
+
+  test('#addRelationships should call uploader enqueue on flushed', async () => {
+    const uploadCollector = createInMemoryStepGraphObjectDataUploaderCollector();
+    const jobState = createTestStepJobStateState({
+      graphObjectStore: new FileSystemGraphObjectStore({
+        graphObjectBufferThreshold: 2,
+      }),
+      uploader: uploadCollector.uploader,
+    });
+
+    const r1 = createTestRelationship();
+    const r2 = createTestRelationship();
+
+    await jobState.addRelationships([r1]);
+    await jobState.addRelationships([r2]);
+
+    const expectedUploaded: FlushedGraphObjectData[] = [
+      {
+        entities: [],
+        relationships: [r1, r2],
+      },
+    ];
+
+    expect(uploadCollector.graphObjectDataCollection).toEqual(expectedUploaded);
+  });
+
+  test('#flush should call uploader enqueue for entities and relationships', async () => {
+    const uploadCollector = createInMemoryStepGraphObjectDataUploaderCollector();
+    const jobState = createTestStepJobStateState({
+      graphObjectStore: new FileSystemGraphObjectStore({
+        graphObjectBufferThreshold: 5,
+      }),
+      uploader: uploadCollector.uploader,
+    });
+
+    const r1 = createTestRelationship();
+    const e1 = createTestEntity();
+
+    await jobState.addRelationships([r1]);
+    await jobState.addEntities([e1]);
+    await jobState.flush();
+
+    const expectedUploaded: FlushedGraphObjectData[] = [
+      {
+        entities: [e1],
+        relationships: [],
+      },
+      {
+        entities: [],
+        relationships: [r1],
+      },
+    ];
+
+    expect(uploadCollector.graphObjectDataCollection).toEqual(expectedUploaded);
+  });
+
+  test('#waitUntilUploadsComplete should resolve when all uploads completed', async () => {
+    const graphObjectDataCollection: FlushedGraphObjectData[] = [];
+
+    const uploader = createQueuedStepGraphObjectDataUploader({
+      stepId: uuid(),
+      uploadConcurrency: 5,
+      async upload(graphObjectData) {
+        await sleep(200);
+        graphObjectDataCollection.push(graphObjectData);
+        return Promise.resolve();
+      },
+    });
+
+    const jobState = createTestStepJobStateState({
+      graphObjectStore: new FileSystemGraphObjectStore({
+        graphObjectBufferThreshold: 2,
+      }),
+      uploader,
+    });
+
+    const e1 = createTestEntity();
+    const e2 = createTestEntity();
+
+    await jobState.addEntities([e1]);
+    await jobState.addEntities([e2]);
+
+    if (!jobState.waitUntilUploadsComplete) {
+      throw new Error(
+        'Default job state should have "waitUntilUploadsComplete" function',
+      );
+    }
+
+    await jobState.waitUntilUploadsComplete();
```

Comment:
This could be a good place to prove to me that `waitUntilUploadsComplete()` is doing its job.
```suggestion
    expect(graphObjectDataCollection).toEqual([]);
    await jobState.waitUntilUploadsComplete();
```